Explorer/Assets/Scripts/Global/Dynamic/PortableExperiences/PortableExperiencesController.cs

```diff
@@ -0,0 +1,180 @@
+                result.configurations.networkId,
+                result.comms?.adapter ?? string.Empty,
+                result.comms?.protocol ?? string.Empty,
+                portableExperiencePath.Value
+            );
+
+            ISceneFacade parentScene = scenesCache.Scenes.FirstOrDefault(s => s.SceneStateProvider.IsCurrent);
+            string parentSceneName = parentScene != null ? parentScene.Info.Name : "main";
+            Entity portableExperienceEntity = world.Create(new PortableExperienceRealmComponent(realmData, parentSceneName, isGlobalPortableExperience), new PortableExperienceComponent(ens));
+            PortableExperienceEntities.Add(ens, portableExperienceEntity);
+
+            WaitToKill(ens).Forget();
+
+            return new IPortableExperiencesController.SpawnResponse
+                { name = realmData.RealmName, ens = ens.ToString(), parent_cid = parentSceneName, pid = portableExperienceEntity.Id.ToString() };
+        }
+
+        private async UniTaskVoid WaitToKill(ENS ens)
+        {
+            await UniTask.Delay(5000);
+            await UnloadPortableExperienceByEnsAsync(ens, CancellationToken.None);
+        }
+
+
+        public bool CanKillPortableExperience(ENS ens)
+        {
+            ISceneFacade currentSceneFacade = scenesCache.Scenes.FirstOrDefault(s => s.SceneStateProvider.IsCurrent);
+            if (currentSceneFacade == null) return false;
+
+            if (PortableExperienceEntities.TryGetValue(ens, out Entity portableExperienceEntity))
+            {
+                PortableExperienceRealmComponent portableExperienceRealmComponent = world.Get<PortableExperienceRealmComponent>(portableExperienceEntity);
+
+                if (portableExperienceRealmComponent.IsGlobalPortableExperience) return false;
+
+                return portableExperienceRealmComponent.ParentSceneId == currentSceneFacade.Info.Name;
+            }
+
+            return false;
+        }
+
+        public List<IPortableExperiencesController.SpawnResponse> GetAllPortableExperiences()
+        {
+            spawnResponsesList.Clear();
+
+            foreach (var portableExperience in PortableExperienceEntities)
+            {
+                PortableExperienceRealmComponent pxRealmComponent = world.Get<PortableExperienceRealmComponent>(portableExperience.Value);
+
+                spawnResponsesList.Add(new IPortableExperiencesController.SpawnResponse {
+                        name = pxRealmComponent.RealmData.RealmName,
+                        ens = portableExperience.Key.ToString(),
+                        parent_cid = pxRealmComponent.ParentSceneId,
+                        pid = portableExperience.Value.Id.ToString() });
+            }
+
+            return spawnResponsesList;
+        }
+
+        public async UniTask<IPortableExperiencesController.ExitResponse> UnloadPortableExperienceByEnsAsync(ENS ens, CancellationToken ct)
```

Comment:
I see no await here. Do we need this as a task?

Reply:
good point!